js/app/demo/app.jsx

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+define(['react', 'jsx!editor', 'jsx!messages', 'jsx!configuration', 'eslint', 'text!../eslint.json'], function(React, Editor, Messages, Configuration, eslint, config) {
+    return React.createClass({
+        displayName: 'App',
+        getInitialState: function() {
+            return {
+                messages: [],
+                options: {
+                    parserOptions: {
+                        ecmaVersion: 5,
+                        sourceType: 'script',
+                        ecmaFeatures: {
+                            jsx: false,
+                            globalReturn: false,
+                            impliedStrict: false,
+                            experimentalObjectRestSpread: false
+                        }
+                    },
+                    rules: JSON.parse(config).rules,
+                    env: {
+                        browser: false,
+                        node: false,
+                        commonjs: false,
+                        "shared-node-browser": false,
+                        worker: false,
+                        amd: false,
+                        mocha: false,
+                        jasmine: false,
+                        jest: false,
+                        phantomjs: false,
+                        jquery: false,
+                        qunit: false,
+                        prototypejs: false,
+                        shelljs: false,
+                        meteor: false,
+                        mongo: false,
+                        protractor: false,
+                        applescript: false,
+                        nashorn: false,
+                        serviceworker: false,
+                        atomtest: false,
+                        embertest: false,
+                        webextensions: false,
+                        es6: false,
+                        greasemonkey: false
+                    }
+                },
+                text: 'var foo = bar;'
+            };
+        },
+
+        handleChange: function(event) {
+            this.setState({ text: event.value }, function() {
+                this.lint();
+            });
+        },
+        updateOptions: function(options) {
+            this.setState({options: options}, function() {
```

Comment:
In general let's avoid the `setState` callback function wherever possibly. It can lead to state bugs and may be causing some of the slowness we're feeling.

What if we just ran the linting on a timer? Every second we run the linting. It doesn't care about events or anything at all. My hunch is that that would help with the performance of everything. We could even have "hasChanged" checks to make sure it's not doing extra work, but I bet it would be way faster, because here we have setState being called inside of setState callbacks and it's probably  causing some silliness